### PR TITLE
(feat) KHP3-6901: Prevent processing of orders with pending bills

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

See https://thepalladiumgroup.atlassian.net/browse/KHP3-6901

This PR represents the first phase of implementing billing functionality across our system. Specifically, it addresses billing concerns within the Labs module. Key points:

1. Implemented billing logic for the Labs module
2. Resolved existing billing-related issues in Labs
3. Laid groundwork for future billing integration

Next steps:

Extend this billing functionality to all relevant order action points throughout the system
cc @FelixKiprotich350 

## Screenshots
Kapture 2024-10-16 at 16 18 00

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Review by Omoshlawi — APPROVED** [review]
Thankz @donaldkibet .Looks good to me

**Comment by ojwanganto** [comment]
Perfect! Thanks, @donaldkibet

File: packages/esm-billing-app/src/billable-services/billiable-item/test-order/test-order-action.resource.tsx [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
 import { BillingConfig } from '../../../config-schema';
+import { usePatientBills } from '../../../modal/require-payment.resource';
+import { useMemo } from 'react';
 
 export const useTestOrderBillStatus = (orderUuid: string, patientUuid: string) => {
   const config = useConfig<BillingConfig>();
   const { currentVisit } = useVisit(patientUuid);
   const { isEmergencyPatient, isLoading: isLoadingQueue } = usePatientQueue(patientUuid);
-  const { isLoading: isLoadingBill, hasPendingPayment } = usePatientBill(orderUuid);
+  const { isLoading: isLoadingBill, hasPendingPayment } = useIsPatientHavingPendingBill(patientUuid, orderUuid);
```
Comment (ojwanganto) [inline packages/esm-billing-app/src/billable-services/billiable-item/test-order/test-order-action.resource.tsx:12]: Do you think we can rename this to indicate an order with unsettled bill? The current name could suggest bills other than for orders. Just my thought

**Review by makombe — DISMISSED** [review]
Thanks @donaldkibet

File: packages/esm-billing-app/src/billable-services/billiable-item/test-order/test-order-action.resource.tsx (hunk 1 above)
Comment (donaldkibet) [reply packages/esm-billing-app/src/billable-services/billiable-item/test-order/test-order-action.resource.tsx:12]: Updated it to `useOrderPendingPaymentStatus`

**Pull request merged**